tracer/src/Datadog.Trace/SpanExtensions.cs

```diff
@@ -119,6 +121,8 @@ public static ISpan SetTag(this ISpan span, string key, double? value)
         [PublicApi]
         public static object GetTagObject(this ISpan span, string key)
```

Comment:
I still really don't like this as an API, do we _have_ to add it? I get customers wanting to add the metric, but do they really need to get it? And if so, could we call this `GetNumericTag(string)` (or something) and _only_ get from metrics, seeing as we already have a `GetTag(string)` method?

Reply:
>  I guess we can remove it and wait for someone asking for it

Personally I'd prefer that, because once we add it, we can never remove it.

I hate public APIs 😛